File: Euler.py
import numpy as np
import matplotlib as plt

a = 0
b = 2
c = 1
f = (lambda x, y: x+y)
N = 100
def Euler(f, a, b, c, N):
    x = np.linspace(a, c, N)
    y = np.zeros(N)
    y[0] = b
    for i in range(N-1):
        y[i+1] = y[i] + f(x[i], y[i])*(x[i+1]-x[i])
    return x, y

"Graficando"
def graficar(x, y):
    plt.plot(x, y, 'r')
    plt.xlabel('x')
    plt.ylabel('y')
    plt.title('Euler')
    plt.show()

File: Rk4_Trabajado_en_clase/Euler.m
% Método de Euler Tradicional

function y=Euler(x,f,y0)
n=length(x);
y=zeros(1,n);
y(1)=y0;
for i=1:n-1
    h=x(i+1)-x(i);
    y(i+1)=y(i)+h*f(x(i),y(i));
end 